src/app/phase/initial-phase/initial-phase.component.ts

```diff
@@ -7,9 +9,31 @@ import { Component, OnInit } from '@angular/core';
 })
 export class InitialPhaseComponent implements OnInit {
 
-  constructor() { }
+  constructor(private docsService:DocsService) { }
 
   ngOnInit(): void {
   }
 
+  getDoc(){
+    let doc = {};
+    console.log(form);
+    doc.id= this.docsService.getNumberOfDocs()+1;
+    doc.name= 'Project Charter';
+    doc.details = {
+
+      projectTitle : document.getElementById('projectTitle').value,
```

Comment:
Based on Tomasz Nurkiewiczs answer, the "problem" is that typescript is typesafe. :) So the document.getElementById() returns the type HTMLElement which does not contain a value property. The subtype HTMLInputElement does however contain the value property.

So a solution is to cast the result of getElementById() to HTMLInputElement like this:

var inputValue = (<HTMLInputElement>document.getElementById(elementId)).value;

Reply:
Reference: https://stackoverflow.com/questions/12989741/the-property-value-does-not-exist-on-value-of-type-htmlelement